### PR TITLE
adds support for aligned crop in search result

### LINKED ISSUE #478: Crop improvements
Currently, the crop can cut in the middle of a word. It would be better if the crop was done at the beginning/end of the word.

Ex: crop of 20 (chars)
```
Hello guys! MeiliSearch <em>is</em> the best search engine I've ever tried. See you soon!
```
is currently cropped this way:
```
o guys! MeiliSearch <em>is</em> the best search e
```
But, it could be cropped this way (=removing the cropped words):
```
guys! MeiliSearch <em>is</em> the best search
```
or this way (=keeping the cropped words):
```
Hello guys! MeiliSearch <em>is</em> the best search engine
```

Suggestion: the `cropLength` could be a number of words instead of a number of characters.

### Version
v8.4.0

### PR DESCRIPTION
**Pull request by MarinPostma**
Adds support for word aligned crops in search result.

fixes #478

### CODE REVIEW

File: meilisearch-http/src/helpers/meilisearch.rs [hunk 1]
```diff
@@ -372,6 +373,29 @@ pub struct SearchResult {
...
 
+/// returns the start index and the length on the crop. 
+fn aligned_crop(text: &str, match_index: usize, context: usize) -> (usize, usize) {
+    if context == 0 {
+        // count need to be at least 1 for cjk queries to return something
+        return (match_index, 1 + text.chars().skip(match_index).take_while(|c| c.is_alphanumeric() && !is_cjk(*c)).count());
+    }
+
+    let word_end_index = |mut index| {
+        if text.chars().nth(index - 1).map_or(false, |c| c.is_alphanumeric()) {
```
Comment (Kerollmops) [inline meilisearch-http/src/helpers/meilisearch.rs]: Not sure but:
```suggestion
        if text.chars().nth(index - 1).map_or(false, |c| c.is_alphanumeric() && is_cjk(*c)) {
```

File: meilisearch-http/src/helpers/meilisearch.rs [hunk 2]
```diff
@@ -372,6 +373,29 @@ pub struct SearchResult {
     pub query: String,
 }
 
+/// returns the start index and the length on the crop. 
+fn aligned_crop(text: &str, match_index: usize, context: usize) -> (usize, usize) {
+    let is_word_component = |c: &char| c.is_alphanumeric() && !is_cjk(*c);
```
Comment (Kerollmops) [inline meilisearch-http/src/helpers/meilisearch.rs]:
```suggestion
    let is_word_component = |c: &char| c.is_alphanumeric() && !is_cjk(*c);

```

Comment (Kerollmops) [reply meilisearch-http/src/helpers/meilisearch.rs]: After that I'll approve and merge :)

**Comment by MarinPostma** [comment]
done!

**Comment by Kerollmops** [comment]
By the way, did you changed the documentation accordingly?

**Pull request merged**